### ISSUE TITLE
Fix warnings

### ISSUE THREAD

**Issue by giampaolo** [pypa/setuptools#1028]
I get this every time I install a package by using `python -Wa setup.py install`. In order to reproduce it pyc files must be deleted first.

```
/home/giampaolo/.local/lib/python3.7/site-packages/setuptools/depends.py:2: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
python3.7 -Wa setup.py build_ext -i
running build_ext
copying build/lib.linux-x86_64-3.7/psutil/_psutil_linux.cpython-37m-x86_64-linux-gnu.so -> psutil
copying build/lib.linux-x86_64-3.7/psutil/_psutil_posix.cpython-37m-x86_64-linux-gnu.so -> psutil
/home/giampaolo/.local/lib/python3.7/site-packages/setuptools/depends.py:2: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
```

**Comment by giampaolo** [comment]
Also, on Python 3.7:

```
/home/giampaolo/.local/lib/python3.7/site-packages/setuptools/command/egg_info.py:115: DeprecationWarning: Flags not at the start of the expression docs/_build/(?:[^/]+ (truncated)
  return re.compile(pat + r'\Z(?ms)')
warning: no previously-included files matching '*' found under directory 'docs/_build'
/home/giampaolo/.local/lib/python3.7/site-packages/setuptools/command/egg_info.py:115: DeprecationWarning: Flags not at the start of the expression \.ci/(?:[^/]+/)*[^/] (truncated)
  return re.compile(pat + r'\Z(?ms)')
```

**Comment by jaraco** [comment]
The imp warning is a duplicate of #479. The second warning was fixed in f1a9711815a.
